activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -252,6 +253,7 @@ def delete_matched(matcher, options = nil)
               begin
                 cursor, keys = node.scan(cursor, match: pattern, count: SCAN_BATCH_SIZE)
                 node.del(*keys) unless keys.empty?
+                keys.each { |key| local_cache.delete_entry(key, merged_options) } if local_cache
```

Comment:
The cache store should not know about the local cache. It is the local cache that should decorate the cache store behavior.

Reply:
I've anticipeated that concern, will try to tackle something workable tomorrow.